target/linux/ipq40xx/image/generic.mk

```diff
@@ -1031,6 +1031,19 @@ define Device/zte_mf286d
 endef
 TARGET_DEVICES += zte_mf286d
 
+define Device/zte_mf289f
```

Comment:
You could extract the parameters shared between MF286D and MF289F, they differ only by DEVICE_MODEL here.

Reply:
Can you explain better?

Thx

Reply:
I`d go with something like:
```
define Device/zte_mf28x_common
	$(call Device/FitzImage)
	DEVICE_VENDOR := ZTE
	SOC := qcom-ipq4019
	DEVICE_DTS_CONFIG := config@ap.dk04.1-c1
	BLOCKSIZE := 128k
	PAGESIZE := 2048
	KERNEL_IN_UBI := 1
	DEVICE_PACKAGES := kmod-usb-net-qmi-wwan kmod-usb-serial-option uqmi
endef

define Device/zte_mf286d
	$(call Device/zte_mf28x_common)
	DEVICE_MODEL := MF286D
endef
TARGET_DEVICES += zte_mf286d

define Device/zte_mf289f
	$(call Device/zte_mf28x_common)
	DEVICE_MODEL := MF289F
endef
TARGET_DEVICES += zte_mf289f
```

Reply:
Nice!

I'll send another commit later :)

Let me know about wifi question